src/Unlock.php

```diff
@@ -9,193 +9,181 @@
 class Unlock
 {
     /**
-     * @param $chat_id1
+     * Add a new row for this channel.
+     *
+     * @param string $chat_id
      *
      * @return bool
      * @throws TelegramException
      */
-    private static function createRow($chat_id1)
+    private static function createRow($chat_id)
     {
         try {
-            $val = 0;
             $pdo = DB::getPdo();
             $sql = 'INSERT INTO `chat_unlock` (status, chat) VALUES (:status, :chat)';
             $sth = $pdo->prepare($sql);
-            $sth->bindParam(':chat', $chat_id1, PDO::PARAM_INT);
-            $sth->bindParam(':status', $val, PDO::PARAM_INT);
-            $sth->execute();
-
-            return self::rowExist($chat_id1);
-
+            $sth->bindValue(':chat', $chat_id);
+            $sth->bindValue(':status', 0, PDO::PARAM_INT);
 
+            return $sth->execute();
         } catch (PDOException $e) {
             throw new TelegramException($e->getMessage());
         }
     }
 
     /**
-     * @param $chat_id2
+     * Check if row exists for this channel.
+     *
+     * @param string $chat_id
      *
      * @return bool
      * @throws TelegramException
      */
-    private static function rowExist($chat_id2)
+    private static function rowExist($chat_id)
     {
         try {
-            $val = '';
             $pdo = DB::getPdo();
-            $sql = 'SELECT * FROM `chat_unlock` WHERE `chat` = :chat';
+            $sql = 'SELECT COUNT(*) FROM `chat_unlock` WHERE `chat` = :chat';
             $sth = $pdo->prepare($sql);
-            $sth->bindParam(':chat', $chat_id2, PDO::PARAM_INT);
+            $sth->bindValue(':chat', $chat_id);
             $sth->execute();
-            $row = $sth->fetch(PDO::FETCH_ASSOC);
-            if (!$row) {
-                $val = false;
-            } elseif ($row) {
-                $val = true;
-            }
-            return $val;
+
+            return $sth->fetchColumn() > 0;
         } catch (PDOException $e) {
             throw new TelegramException($e->getMessage());
         }
-
     }
 
     /**
-     * @param $chat_id3
+     * Check if this channel is unlocked.
      *
-     * @return bool|string
+     * @param string $chat_id
+     *
+     * @return bool
+     * @throws TelegramException
      */
-    public static function isUnlocked($chat_id3)
+    public static function isUnlocked($chat_id)
     {
-        if (!self::rowExist($chat_id3)) {
-            if (!self::createRow($chat_id3)) {
-                return false;
-            }
+        if (!self::rowExist($chat_id) && !self::createRow($chat_id)) {
```

Comment:
If i'm right it would trigger the createRow function everytime and it would create one. Thats why there was two if statement so the createRow gets only triggert when there is no row. could combine those two funtion to one

Reply:
It's exactly the same thing. You just had it nested, but in effect it does the same. It only returns `false` if both conditions apply. So if the row exists, it doesn't call `self::createRow(...)`.